com.unity.ml-agents.extensions/Runtime/Sensors/GridSensor.cs

```diff
@@ -914,10 +906,22 @@ void ISensor.Update()
 
         /// <summary>Gets the observation shape</summary>
         /// <returns>int[] of the observation shape</returns>
+        public ObservationSpec GetObservationSpec()
+        {
+            // Lazy update
+            var shape = m_ObservationSpec.Shape;
+            if (shape[0] != GridNumSideX || shape[1] != GridNumSideZ || shape[2] != ObservationPerCell)
+            {
+                m_ObservationSpec = ObservationSpec.Visual(GridNumSideX, GridNumSideZ, ObservationPerCell);
+            }
+            return m_ObservationSpec;
+        }
+
+        /// <inheritdoc/>
         public override int[] GetObservationShape()
         {
-            m_Shape = new[] { GridNumSideX, GridNumSideZ, ObservationPerCell };
-            return m_Shape;
+            var shape = m_ObservationSpec.Shape;
```

Comment:
This is the SensorComponent override, not sure if we should change that to return a spec too.

Reply:
I would lean toward yes

Reply:
OK, I'm going to punt that to a separate PR.